test/integration/src/test/java/io/pravega/test/integration/ReaderGroupNotificationTest.java

```diff
@@ -265,5 +398,57 @@ public void testEndOfStreamNotifications() throws Exception {
         listenerLatch.await();
         assertTrue("Listener invoked", listenerInvoked.get());
     }
+    
+    private List<List<Segment>> scale(Controller controller, String scopeName, String streamName, int numSegments, int scalesToPerform, ScheduledExecutorService executorService) {
+        Stream stream = new StreamImpl(scopeName, streamName);
+        AtomicInteger counter = new AtomicInteger(0);
+        List<List<Segment>> listOfEpochs = new LinkedList<>();
+
+        CompletableFuture<Void> scaleFuture = Futures.loop(() -> counter.incrementAndGet() <= scalesToPerform,
+                () -> controller.getCurrentSegments(scopeName, streamName)
+                        .thenCompose(segments -> {
+                            ArrayList<Segment> sorted = Lists.newArrayList(segments.getSegments().stream()
```

Comment:
1. Why do you make a copy of a list that you have just created and you only use for one purpose?
2. Why do you use `Lists.newArrayList`? What's wrong with `new ArrayList<>(...)`?
3. I don't understand what this sorting is trying to achieve.